### ISSUE TITLE
Style tag inside svelte:head marked as an error

### ISSUE THREAD

**Issue by ehrencrona** [sveltejs/language-tools#586]
The following file does not check out with `svelte-check` (I noticed this while looking at sveltejs/prettier-plugin-svelte#149):

```
<svelte:head>
  <style>
  </style>
</svelte:head>
```

```
> npx svelte-check@1.0.54 src/deleteme.svelte 

Loading svelte-check in workspace: /projects/svelte/prettier-plugin-svelte
Getting Svelte diagnostics...
====================================

/projects/svelte/prettier-plugin-svelte/src/deleteme.svelte:5:17
Error: ';' expected. (js)
</svelte:head>

/projects/svelte/prettier-plugin-svelte/src/deleteme.svelte:6:1
Error: Expression expected. (js)
```

**System (please complete the following information):**
MacOS, Node 14.4, Svelte-check 1.0.54

**Comment by dummdidumm** [comment]
`svelte2tsx` transformation is wrong, gets transformed to

```
///<reference types="svelte" />
<></>;function render() {
<><sveltehead>
  
</sveltehead> </></style> // <---------------- wrong
return { props: {}, slots: {}, getters: {}, events: {} }}

export default class extends createSvelte2TsxComponent(__sveltets_partial(__sveltets_with_any_event(render))) {
}
```

**Comment by ehrencrona** [comment]
It's this line of code https://github.com/sveltejs/language-tools/blob/master/packages/svelte2tsx/src/htmlxtojsx/nodes/element.ts#L20 together with the issue I fixed in sveltejs/svelte#5487 

That was tricky to find :D
